feat: add personal builder page

gboigwe opened the pull request:
## Description

Added some information about me

## Additional Information

- [x] I have read the [contributing docs](/scaffold-eth/scaffold-eth-2/blob/main/CONTRIBUTING.md) (if this is your first contribution)
- [x] This is not a duplicate of any [existing pull request](https://github.com/scaffold-eth/scaffold-eth-2/pulls)

## Related Issues

_Closes #{2}_

Your ENS/address: 0x5cc8Be96B1C9A68F57a73b5bEa60cF5D890055A1

phipsae commented on packages/nextjs/app/builders/0x5cc8Be96B1C9A68F57a73b5bEa60cF5D890055A1/page.tsx:
```diff
@@ -0,0 +1,85 @@
+  socialIcon: "p-2 hover:bg-base-200 rounded-full transition-colors duration-200 w-10 h-10",
+};
+
+const BuilderPage = () => {
```
You might want to give your page a more descriptive name (we will have more builder pages) and consider typing your page as NextPage since we are using TypeScript, as this ensures better integration and adds clarity for future maintainers by indicating that the component is a Next.js page.

phipsae commented on packages/nextjs/app/builders/0x5cc8Be96B1C9A68F57a73b5bEa60cF5D890055A1/page.tsx:
```diff
@@ -0,0 +1,85 @@
+};
+
+const BuilderPage = () => {
+  const builderDetails = {
```
You could place the const outside of the component since it is a static profile page.

phipsae commented on packages/nextjs/app/builders/0x5cc8Be96B1C9A68F57a73b5bEa60cF5D890055A1/page.tsx:
```diff
@@ -0,0 +1,85 @@
+  const builderDetails = {
+    name: "Gbolahan Akande",
+    address: "0x5cc8Be96B1C9A68F57a73b5bEa60cF5D890055A1",
+    bio: "A passionate Web3 developer with experience in Solidity and full-stack development. Currently building decentralized applications and contributing to BuidlGuidl projects. Excited about the future of blockchain technology and its potential to reshape the digital world.",
```
Some highlighting in the text might be nice.

phipsae commented on packages/nextjs/app/builders/0x5cc8Be96B1C9A68F57a73b5bEa60cF5D890055A1/page.tsx:
```diff
@@ -0,0 +1,85 @@
+          <img src={builderDetails.avatar} alt="Profile" className={styles.avatar} />
+          <div className={styles.nameSection}>
+            <h1 className={styles.name}>{builderDetails.name}</h1>
+            <p className={styles.address}>{builderDetails.address}</p>
```
You could consider using the SE2 component `<Address>`

phipsae reviewed (changes requested):
Thanks a lot for the page! 

Below you'll find some little comments.

gboigwe commented:
# Personal Builder Page

## Description
Added my personal builder page with the following features:
- Responsive design for mobile and desktop
- Dark/light mode support
- Enhanced typography and visual hierarchy
- Social links integration
- Address display using SE2 components

## Screenshots
![image](https://github.com/user-attachments/assets/9523f7a0-2113-472f-ac99-097362065f4c)

## Additional Notes
- Implemented all feedback from previous review
- Used existing Scaffold-ETH 2 components
- Maintained consistent styling with the project

gboigwe commented:
# Enhanced Homepage and Builders Grid

## Changes Made
- ✨ Improved homepage dark mode visibility and contrast
- 🏗️ Added BuildersGrid component showing registered builders
- 🔄 Created API route to fetch builder addresses from filesystem
- 🔗 Connected grid to personal builder pages (/builders/0x...)
- 💅 Updated styling and layout for better UX

## Implementation Details
- Homepage now reads checkedInCounter from BatchRegistry contract
- BuildersGrid scans filesystem for builder pages in `/builders` directory
- Added loading states and error handling
- Maintained consistent styling with existing components

## Screenshots
### The checked in count:
![image](https://github.com/user-attachments/assets/270c7511-9767-4704-8459-85d92fd6851c)

### The builders profile grid:
![image](https://github.com/user-attachments/assets/0605b088-34bb-4c06-b987-fa49fe60a0b1)
- For now I only have mine, but if this PR is accepted and merged, then every builder would be registered in.

## Testing Done
- Tested dark mode visibility
- Verified builder count display
- Confirmed grid updates when new builder pages are added
- Checked mobile responsiveness

## Additional Notes
- Grid automatically updates as builders add their profile pages
- Uses existing Address component for consistency

phipsae commented on packages/nextjs/app/page.tsx:
I know in the beginning it is a little bit confusing. For each and every issue you tackle there should be a separate PR and branch. So it's easier for reviewers to understand what's going on. 

And this isn't related to your personal page PR, so please leave that out.

gboigwe replied:
I have just reset and forced push up to only my personal page

phipsae commented on packages/nextjs/components/batch/BuildersGrid.tsx:
I know in the beginning it is a little bit confusing. For each and every issue you tackle there should be a separate PR and branch. So it's easier for reviewers to understand what's going on. 

And this isn't related to your personal page PR, so please leave that out.

gboigwe replied:
I have just reset and forced pushed up to only my personal page

phipsae commented on packages/nextjs/app/api/builders/route.ts:
I know in the beginning it is a little bit confusing. For each and every issue you tackle there should be a separate PR and branch. So it's easier for reviewers to understand what's going on. 

And this isn't related to your personal page PR, so please leave that out.

gboigwe replied:
I have just reset and forced pushed up to only my personal page

phipsae reviewed (changes requested):
Thanks for the changes and also the implementation of builders overview!
 
Unfortunately that has nothing to do with your personal page PR, pls take that out and create a dedicated PR for that. 

Thanks a lot!

gboigwe commented:
Please confirm now that the PR has only solved the issue pertaining to it. Thank you

phipsae commented:
Perfect, no its good, thanks! Lets merge!

gboigwe commented:
> Perfect, no its good, thanks! Lets merge! 

Thank you, It's a real great relief comment. Thanks.
